lib/services/plugins-service.ts

```diff
@@ -12,23 +12,34 @@ export class PluginsService implements IPluginsService {
 	private static CORE_PLUGINS_PROPERTY_NAME = "CorePlugins";
 	private static CORDOVA_PLUGIN_VARIABLES_PROPERTY_NAME = "CordovaPluginVariables";
 	private static MESSAGES = ["Core Plugins", "Advanced Plugins", "Marketplace Plugins"];
-	private identifierToPlugin: IDictionary<IPlugin>;
+	private _identifierToPlugin: IDictionary<IPlugin>;
+	private get identifierToPlugin(): IDictionary<IPlugin> {
+		if (!this._identifierToPlugin) {
+			this.loadPluginsData().wait();	
+		}
+
+		return this._identifierToPlugin;
+	}
 
-	constructor($cordovaPluginsService: ICordovaPluginsService,
-		$marketplacePluginsService: ICordovaPluginsService,
+	constructor(private $cordovaPluginsService: ICordovaPluginsService,
+		private $marketplacePluginsService: ICordovaPluginsService,
 		private $errors: IErrors,
 		private $logger: ILogger,
 		private $project: Project.IProject,
 		private $prompter: IPrompter,
 		private $loginManager: ILoginManager,
 		private $projectConstants: Project.IProjectConstants) {
+	}
 
-		// Cordova plugin commands are only applicable to Cordova projects
-		this.$project.ensureCordovaProject();
-		this.$loginManager.ensureLoggedIn().wait();
-		this.identifierToPlugin = Object.create(null);
-		Future.wait([this.createPluginsData($cordovaPluginsService),
-			this.createPluginsData($marketplacePluginsService)]);
+	private loadPluginsData(): IFuture<void> {
+		return (() => {
+			// Cordova plugin commands are only applicable to Cordova projects
+			this.$project.ensureCordovaProject();
+			this.$loginManager.ensureLoggedIn().wait();
+			this._identifierToPlugin = Object.create(null);
+			Future.wait([this.createPluginsData(this.$cordovaPluginsService),
```

Comment:
createPluginsData can throw, are we OK with Future.wait() swallowing the exception?